ably-ios/ARTRest.m

```diff
@@ -200,10 +207,20 @@ - (void)time:(void(^)(NSDate *time, NSError *error))callback {
     return nil;
 }
 
-- (void)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback {
-    NSParameterAssert(query.limit < 1000);
-    NSParameterAssert([query.start compare:query.end] != NSOrderedDescending);
-    
+- (BOOL)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback error:(NSError **)errorPtr {
+    if (query.limit > 1000 && errorPtr) {
```

Comment:
https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/ErrorHandlingCocoa/CreateCustomizeNSError/CreateCustomizeNSError.html#//apple_ref/doc/uid/TP40001806-CH204-BAJIIGCC

On 24 December 2015 at 18:59:41, Toni Cárdenas (notifications@github.com) wrote:

In ably-ios/ARTRest.m:

> @@ -200,10 +207,20 @@ - (void)time:(void(^)(NSDate *time, NSError *error))callback {
>      return nil;
>  }
>   
> -- (void)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback {
> -    NSParameterAssert(query.limit < 1000);
> -    NSParameterAssert([query.start compare:query.end] != NSOrderedDescending);
> -     
> +- (BOOL)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback error:(NSError **)errorPtr {
> +    if (query.limit > 1000 && errorPtr) {
We need to return NO even when errorPtr is nil, otherwise we will make an invalid request.

—
Reply to this email directly or view it on GitHub.